pkg/kn/commands/revision_get.go

```diff
@@ -65,3 +76,38 @@ func NewRevisionGetCommand(p *KnParams) *cobra.Command {
 	revisionGetFlags.AddFlags(revisionGetCommand)
 	return revisionGetCommand
 }
+
+func getRevisionInfo(args []string, namespace string, client serving.ServingV1alpha1Interface, cmd *cobra.Command) (interface{}, error) {
+
+	switch len(args) {
+	case 0:
+		revision, err := client.Revisions(namespace).List(v1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		if len(revision.Items) == 0 {
+			fmt.Fprintf(cmd.OutOrStdout(), "No resources found.\n")
+			return nil, nil
+		}
+		revision.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "knative.dev",
+			Version: "v1alpha1",
+			Kind:    "revision"})
+		return revision, nil
+
+	case 1:
+		revision, err := client.Revisions(namespace).Get(args[0], v1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		revision.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "serving.knative.dev",
+			Version: "v1alpha1",
+			Kind:    "revision"})
+
+		return revision, nil
+	default:
+		return nil, errors.New("revision get does not accept more than one NAME\nUsage: kn revision get NAME")
```

Comment:
I don't think errors should contain any formatting characters. It's not about the inner command code which knows in which context it is used. Imaging that this command is used in a kind of library, then printing out the usage information here is for sure the wrong place. Just return the error without any implicit knowledge where this function could be used, but add as much context information that you have available. Like the revisions names provided and detected.

Reply:
Maybe something like

```
fmt.Errorf("'revision get only accepts no arguments or a single revision name, not %d arguments as given",len(args))
```

Reply:
I will change the message thanks